CMakeLists.txt

```diff
@@ -10,10 +10,10 @@ option(EXPOSE_EXTERNAL_PAGEMAP "Expose the global pagemap" OFF)
 option(EXPOSE_EXTERNAL_RESERVE "Expose an interface to reserve memory using the default memory provider" OFF)
 option(SNMALLOC_RUST_SUPPORT "Build static library for rust" OFF)
 option(SNMALLOC_QEMU_WORKAROUND "Disable using madvise(DONT_NEED) to zero memory on Linux" Off)
-option(SNMALLOC_ANDROID "Set cross build target to Android" OFF)
+option(SNMALLOC_ARM_ANDROID "Set cross build target to ARM Android" OFF)
```

Comment:
The reason why I change this is that android does not always mean `arm`

Reply:
Why is this needed separately from setting `CMAKE_SYSTEM_PROCESSOR`?  Normally, when you cross-compile with CMake, you provide a CMake toolchain file that sets things like this.  A quick search suggests that the Android NDK comes with one of these (apparently it's in `<NDK>/build/cmake/android.toolchain.cmake`).  Does that not set the correct things?

Reply:
You are right. `CMAKE_TOOLCHAIN_FILE` is definitely a good choice, will look into it later.

I set this just because passing `CMAKE_SYSTEM_PROCESSOR` via command line is not working. 

It should be set by `android.toolchain.cmake`.

Reply:
using toolchain file for `androidebi` will solve the linking issue too. the linking result of `emutls` after changing to the new way:
```
0002d7e0 t __emutls_get_address
0002d8b8 t __emutls_register_common
0003eff4 r __emutls_t._ZL16g_current_locale
0003e15c r __emutls_t._ZN10__cxxabiv112_GLOBAL__N_15dtorsE
00040dc0 V __emutls_t._ZZN8snmalloc17ThreadAllocCommon13get_referenceEvE5alloc
0002d7c8 t __emutls_unregister_key
00043020 V __emutls_v._ZGVZN8snmalloc27ThreadAllocThreadDestructor16register_cleanupEvE6tidier
00043080 d __emutls_v._ZL16g_current_locale
00043050 d __emutls_v._ZN10__cxxabiv112_GLOBAL__N_111dtors_aliveE
00043060 d __emutls_v._ZN10__cxxabiv112_GLOBAL__N_15dtorsE
00043030 V __emutls_v._ZN8snmalloc17ThreadAllocCommon18destructor_has_runE
00043040 V __emutls_v._ZN8snmalloc19operation_in_flightE
00043000 V __emutls_v._ZZN8snmalloc17ThreadAllocCommon13get_referenceEvE5alloc
00043010 V __emutls_v._ZZN8snmalloc27ThreadAllocThreadDestructor16register_cleanupEvE6tidier
0002d6ec t emutls_alloc
0002d6a8 t emutls_destroy
0002d784 t emutls_init
00046dd8 b emutls_key
00046de0 b emutls_key_created
00046ddc b emutls_mutex
00046de8 b emutls_size
```